app/views/generic_object_definition/_show_list.html.haml

```diff
@@ -7,9 +7,12 @@
     = render :partial => 'generic_object_definition/show_actions'
   - if @custom_button_group_node
     = render :partial => 'generic_object_definition/show_custom_button_group'
+  - if @custom_button_node
+    = render :partial => 'generic_object_definition/show_custom_button'
 
   %generic-object-definition-toolbar#generic_object_definition_show_list_form{'entity' => _('Generic Object Class')}
 
 :javascript
   miq_bootstrap('#generic_object_definition_show_list_form');
+  $('#searchbox').hide();
```

Comment:
This should not be needed after https://github.com/ManageIQ/manageiq-ui-classic/pull/2635

Reply:
The only reason I added that was to hide the `searchbox` in case of a reload on a `show custom group` page or a `show button` page.
Ideally I would like to get rid of it with a more valid approach if there is any.

Reply:
Unfortunately, #2635 does not work if I reload the page (as I mentioned above)

Please suggest if you know of a better approach to not make that box appear.
Is there an instance variable that needs to be set? I don't know.

Reply:
Awaiting your reply here if there is a better way to hide the search box on show screens on a reload.

Reply:
Ok, sure.. so, where does `#searchbox` come from...

```sh
$ ag '#searchbox'
app/views/layouts/_x_adv_searchbox.html.haml
10:  #searchbox.search-pf.has-button.form-group.pull-right{:class => ::Settings.server.custom_logo ? "whitelabeled" : ""}

app/views/layouts/_searchbar.html.haml
6:  #searchbox
```

So one of the 2 .. trying which one, I can see it's the latter... let's look at it..

```haml
- if @lastaction == "show_list" && !session[:menu_click] && display_adv_search? && !@in_a_form
  ...
  #searchbox
```

So.. I guess all of those are true, including `display_adv_search?` .. which contains a long list of layouts to display advanced search for.

So you may just need to add a condition there... but most likely the problem is that those screens should not have the `generic_object_definition` `@layout` (or that layout should not be mentioned in that whitelist).

Reply:
I think we are not left with any other alternatives here.
Hence we would have to just go with the `.hide()` solution.

Reply:
Great, so.. can't you use a different `layout` for this screen?

Reply:
What I currently have is the standard implementation.

It's best to leave the standard implementation and all the default settings like `layout` that go with it, as-is.

In some hamls, I have noticed that we do this in order to show the searchbox -
```ruby
:javascript
  ManageIQ.afterOnload = "if (miqDomElementExists('adv_searchbox_div')) $('#adv_searchbox_div').show();"
```

So looks like the jquery `show()/hide()`, is the only clean way to show/hide the searchbox in treeviews

Reply:
Created this - 1ce6c4f 
It does the show/hide based on what haml is being displayed in the right cell.